File: test-suite/src/test/java/io/micronaut/docs/factories/nullable/EngineConfiguration.java
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.docs.factories.nullable;

import io.micronaut.context.annotation.EachProperty;
import io.micronaut.core.util.Toggleable;

import javax.validation.constraints.NotNull;

// tag::class[]
@EachProperty("engines")
public class EngineConfiguration implements Toggleable {

    private boolean enabled = true;
    private Integer cylinders;

    @NotNull
    public Integer getCylinders() {
        return cylinders;
    }

    public void setCylinders(Integer cylinders) {
        this.cylinders = cylinders;
    }

    @Override
    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

}
// end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/factories/nullable/EngineConfiguration.kt
package io.micronaut.docs.factories.nullable

import io.micronaut.context.annotation.EachProperty
import io.micronaut.core.util.Toggleable
import javax.validation.constraints.NotNull

// tag::class[]
@EachProperty("engines")
class EngineConfiguration : Toggleable {

    val enabled = true

    @NotNull
    val cylinders: Int? = null

    override fun isEnabled(): Boolean {
        return enabled
    }
}
// end::class[]
